nomos-core/src/block/builder.rs

```diff
@@ -0,0 +1,100 @@
+// std
+use serde::de::DeserializeOwned;
+use serde::Serialize;
+use std::hash::Hash;
+use std::marker::PhantomData;
+// crates
+
+// internal
+use crate::block::Block;
+use crate::da::blob::{Blob, BlobSelect};
+use crate::tx::{Transaction, TxSelect};
+use consensus_engine::overlay::RandomBeaconState;
+use consensus_engine::{NodeId, Qc, View};
+
+pub struct BlockBuilder<Tx, Blob, TxSelector, BlobSelector> {
```

Comment:
how would this be used? I think an example might help me understand

Reply:
I'm not really sure I understand exactly. But it is just a wrapper over the `Block::new` that throws into the mix the transaction and blob selectors (Which should be some static heuristic). 

Core is the build method:

```rust
Block::new(
                view,
                parent_qc,
                tx_selector.select_tx_from(txs),
                blob_selector.select_blob_from(blobs),
                proposer,
                beacon,
            )
```

The idea is to just use it when the leader propose the block.

I had in mind that we could even have a service that handles the next block building depending on different events in the node:
* View change
* New tx arrived
* New blob arrived
* ...
So it always have something ready when needed. 

But for now I think this is enough.

Reply:
> how would this be used? I think an example might help me understand

I added an example to the builder, but I do not think that is what you reffered to, hence my answer above.